parallel.go

```diff
@@ -289,20 +289,57 @@ func (r Parallel) forKey(key string) Parallel {
 	})
 }
 
+// mergeQueryEvents limits `routing.QueryError` events to only be sent on the context in case all parallel
+// routers fail.
+func (r Parallel) mergeQueryEvents(ctx context.Context) (context.Context, context.CancelFunc) {
+	subCtx, evCh := routing.RegisterForQueryEvents(ctx)
+	go func() {
+		var errEvt *routing.QueryEvent
+		successfulEvent := false
+		for {
+			select {
+			case <-subCtx.Done():
+				if errEvt != nil && !successfulEvent {
+					routing.PublishQueryEvent(ctx, errEvt)
+				}
+				return
+			case ev, ok := <-evCh:
+				if !ok {
+					return
+				}
+				if ev == nil {
+					continue
+				}
+				if ev.Type == routing.QueryError {
+					errEvt = ev
+					continue
+				}
+				successfulEvent = true
+				routing.PublishQueryEvent(ctx, ev)
+			}
+		}
+	}()
+	return context.WithCancel(subCtx)
```

Comment:
?

Reply:
Ah, sorry. I think we still want to create a cancelable context and make sure to cancel it. My point was that we should create the cancelable context _before_ registering for events.